src/quo2/components/wallet/network_routing/view.cljs

```diff
@@ -0,0 +1,171 @@
+(ns quo2.components.wallet.network-routing.view
+  (:require
+    [oops.core :as oops]
+    [quo2.components.wallet.network-routing.animation :as animation]
+    [quo2.components.wallet.network-routing.style :as style]
+    [quo2.theme :as quo.theme]
+    [react-native.core :as rn]
+    [react-native.gesture :as gesture]
+    [react-native.reanimated :as reanimated]
+    [reagent.core :as reagent]
+    [utils.number]))
+
+(defn- slider
+  [slider-shared-values]
+  [rn/view {:style style/slider-container}
+   [reanimated/view {:style (style/slider slider-shared-values)}]])
+
+(defn network-bar
+  [_]
+  (let [detecting-gesture?      (reagent/atom false)
+        amount-on-gesture-start (atom 0)]
+    (fn [{:keys                [total-width total-amount on-press on-new-amount allow-press?]
+          {:keys [amount-shared-value
+                  max-amount]} :bar
+          :as                  props}]
+      (let [slider-width-shared-value   (reanimated/use-shared-value 4)
+            slider-height-shared-value  (reanimated/use-shared-value 32)
+            slider-opacity-shared-value (reanimated/use-shared-value 0)
+            width->amount               #(/ (* % total-amount) total-width)]
+        [rn/touchable-without-feedback
+         {:on-press #(when (and (not @detecting-gesture?) allow-press?)
+                       (on-press)
+                       (reset! detecting-gesture? true)
+                       (animation/show-slider slider-opacity-shared-value))}
+         [reanimated/view {:style (style/network-bar props)}
+          [gesture/gesture-detector
+           {:gesture
+            (-> (gesture/gesture-pan)
+                (gesture/enabled @detecting-gesture?)
+                (gesture/on-begin
+                 (fn [_]
+                   (animation/increase-slider slider-width-shared-value slider-height-shared-value)
+                   (reset! amount-on-gesture-start (reanimated/get-shared-value amount-shared-value))))
+                (gesture/on-update
+                 (fn [event]
+                   (let [new-amount (-> (oops/oget event "translationX")
+                                        (width->amount)
+                                        (+ @amount-on-gesture-start)
+                                        (utils.number/value-in-range 1 max-amount))]
+                     (reanimated/set-shared-value amount-shared-value new-amount))))
+                (gesture/on-finalize
+                 (fn [_]
+                   (animation/decrease-slider slider-width-shared-value slider-height-shared-value)
+                   (animation/hide-slider slider-opacity-shared-value)
+                   (on-new-amount (reanimated/get-shared-value amount-shared-value))
+                   (reset! detecting-gesture? false))))}
```

Comment:
Maybe we can move this code to a separate function?

Reply:
I was thinking about that too.
But that function will require a lot of arguments, I realized the code was wasier to read/track by placing the logic here.

Reply:
what if they are broken up into smaller functions? e.g a defn for on-finalize, for on-update etc

Reply:
i.e
```
(-> (gesture/gesture-pan)
                (gesture/enabled @detecting-gesture?)
                (gesture/on-begin (begin ... .... ...)
                (gesture/on-update (update ... .... ....))
                (gesture/on-finalize (finalize ... .... ...))))}
 ```

Reply:
Yep, also tried, and also tried place them in the `animated` namespace I created.

At the end I realized the gesture could live here. Extracting these functions require to receive multiple parameters and one thing I dont like: passing the `detecting-gesture?` atom.

As you can see, at the end of the `on-finalize` function, a `reset!` on an atom is performed (as well as `on-begin`). So, if we want to extract the logic to a separate function, we have two options:

1. Pass the atom and perform the reset! inside the extracted function:
```clojure
(animation/on-finalize {:detecting-gesture-atom detecting-gesture? ,,,}
```
Which I find too dirty, because we are passing a reactive atom and we would be making the code more complex, i.e. now the extracted function must know that `detecting-gesture?` is an atom that could be `deref`ed, `reset!` and `swap!`.

2.  Pass a callback:
```clojure
(animation/on-finalize {:on-finalize (fn [] (reset! detecting-gesture? false))}
```
which is more clean in terms that we are not passing the atom. but I think it's redundant.

IMO keeping the gesture code  in the view is not  too bad because is not too hard to understand/follow. the heavy work is done in `animated` namespace.

WDYT?

Reply:
Sure, no strong opinions from my side. Whatever you think is good for me 👍